internal/storage/storageutil/custom_retry_test.go

```diff
@@ -0,0 +1,70 @@
+package storageutil
+
+import (
+	"errors"
+	"io"
+	"net"
+	"net/url"
+	"testing"
+
+	. "github.com/jacobsa/ogletest"
+	"google.golang.org/api/googleapi"
+)
+
+func TestCustomRetry(t *testing.T) { RunTests(t) }
+
+type customRetryTest struct {
+}
+
+func init() { RegisterTestSuite(&customRetryTest{}) }
+
+func (t customRetryTest) ShouldRetryReturnsTrueWithGoogleApiError() {
+	// 401
+	var err401 = googleapi.Error{
+		Code: 401,
+		Body: "Invalid Credential",
+	}
+	// 50x error
+	var err502 = googleapi.Error{
+		Code: 502,
+	}
+	// 429 - rate limiting error
+	var err429 = googleapi.Error{
+		Code: 429,
+		Body: "API rate limit exceeded",
+	}
+
+	ExpectEq(ShouldRetry(&err401), true)
+	ExpectEq(ShouldRetry(&err502), true)
+	ExpectEq(ShouldRetry(&err429), true)
+}
+
+func (t customRetryTest) ShouldRetryReturnsFalseWithGoogleApiError() {
```

Comment:
isn't this conflicting with above test case
say here ShouldRetryReturnsFalseWithGoogleApiError400